dataprofiler/profilers/profile_builder.py

```diff
@@ -602,37 +604,32 @@ def update_profile(self, data, sample_size=None, min_true_samples=None):
         :type min_true_samples
         :return: None
         """
-        if not sample_size:
-            sample_size = self._samples_per_update
         if not min_true_samples:
             min_true_samples = self._min_true_samples
         if isinstance(data, data_readers.base_data.BaseData):
-            self._profile = self._update_profile_from_chunk(
-                data.data, self._profile, sample_size, min_true_samples, self.options)
+            self._update_profile_from_chunk(
+                data.data, sample_size, min_true_samples, self.options)
             self._update_row_statistics(data.data)
             self.encoding = data.file_encoding
             self.file_type = data.data_type
         elif isinstance(data, pd.DataFrame):
-            self._profile = self._update_profile_from_chunk(
-                data, self._profile, sample_size, min_true_samples, self.options)
+            self._update_profile_from_chunk(
+                data, sample_size, min_true_samples, self.options)
```

Comment:
previously `_update_profile_from_chunk` was static, but it doesn't really make sense and means we no longer have to pass profiles and can just use `self._profile` within the function

Reply:
This was done so that `_get_sample_size` could be called within `_update_profile_from_chunk`. If we did it here, the function would have to be rearranged in order to account for data checks which previously were not needed.

Currently the data check is the `else` in the conditionals, but if sample size was acquired here, it would need to move to the top  followed by the `_get_sample_size` and the instance check conditionals. Feasible if desired.

Reply:
It's not clear to me why we need to call `self._get_sample_size` within the `_update_profile_from_chunk`, we already pass in `sample_size` here.